agent/engine/task_manager.go

```diff
@@ -285,8 +285,9 @@ func (mtask *managedTask) waitForHostResources() {
 		return
 	}
 
-	if !mtask.IsInternal && !mtask.engine.hostResourceManager.checkTaskConsumed(mtask.Arn) {
+	if !mtask.IsLaunchTypeFargate() && !mtask.IsInternal && !mtask.engine.hostResourceManager.checkTaskConsumed(mtask.Arn) {
```

Comment:
Is this the only change we need? The PR description says "Their resources never get accounted in HostResourceManager.". What prevents a Fargate task's resources from being accounted in HostResourceManager?

Reply:
All tasks have a managedTask, and go through [overseeTask](https://github.com/aws/amazon-ecs-agent/blob/08f9dc120c9a23c3e5c7f969e48398a243066819/agent/engine/task_manager.go#L197) on start up. It is a linear path to here on receiving a Start task request on an already started instance. So this should be the only needed change.

This change prevents queuing up task to go through [waitingTaskQueue](https://github.com/aws/amazon-ecs-agent/blob/08f9dc120c9a23c3e5c7f969e48398a243066819/agent/engine/docker_task_engine.go#L392-L413) processing for Fargate tasks, and hence they will never be accounted by HostResourceManager

Reply:
Thanks!